### PR TITLE
Feature: Add GitHub action workflow

### PR DESCRIPTION
**Pull request by maxhoheiser**
Add a GitHub action workflow, that installs required dependencies, starts a Localstack container, deploys the infrastructure and runs the pytest

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      - 'README.md'
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  schedule:
+    # “At 00:00 on Sunday.”
+    - cron: "0 0 * * 0"
+  workflow_dispatch:
```
Comment (joe4dev) [inline .github/workflows/main.yaml:12]: not needed?

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        with:
+            image-tag: ${{ inputs.release-tag || 'latest'}}
+            use-pro: 'true'
+            install-awslocal: 'true'
+            configuration: DEBUG=1
+        env:
+            LOCALSTACK_API_KEY: ${{ secrets.LOCALSTACK_API_KEY }}
+
+      - name: Deploy using CDK
+        run: |
```
Comment (joe4dev) [inline .github/workflows/main.yaml:68]: Consider invoking the make targets here

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            cdklocal bootstrap
+            cdklocal synth
+            echo "Deploy redshift stack"
+            cdklocal deploy KinesisFirehoseRedshiftStack1 --require-approval never
+            python utils/prepare_redshift.py
+            echo "Deploy firehose stack"
+            cdklocal deploy KinesisFirehoseRedshiftStack2 --require-approval never
+
+      - name: List the resources
+        run: |
```
Comment (joe4dev) [inline .github/workflows/main.yaml:72]: e.g., `make list_resources`

File: .github/workflows/main.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            awslocal s3 ls
+            awslocal kinesis list-streams
+            awslocal firehose list-delivery-streams
+            awslocal redshift describe-clusters
+
+      - name: Run pytests
+        run: |
```
Comment (joe4dev) [inline .github/workflows/main.yaml:76]: `make test`

File: .github/workflows/main.yaml (hunk 2 above)
Comment (joe4dev) [reply .github/workflows/main.yaml:68]: but that makes it a bit trickier to control the exit codes. Chaining with `&& \` instead of newlines would do it

**Review by joe4dev — CHANGES_REQUESTED** [review]
It looks like the workflow is failing with the error `TypeError: type of argument db_name must be str; got NoneType instead`

Is this waiting for a dependent PR?

File: .github/workflows/main.yaml (hunk 1 above)
Comment (maxhoheiser) [reply .github/workflows/main.yaml:12]: deleted in: https://github.com/localstack-samples/sample-cdk-kinesis-firehose-redshift/pull/3/commits/c7bb4b90b9906faeceeb60e07b5bbc03ab7161d0

File: .github/workflows/main.yaml (hunk 3 above)
Comment (maxhoheiser) [reply .github/workflows/main.yaml:72]: fixed in: https://github.com/localstack-samples/sample-cdk-kinesis-firehose-redshift/pull/3/commits/d590f1498acf26c3ccb669aaac0740957b5d4355

File: .github/workflows/main.yaml (hunk 4 above)
Comment (maxhoheiser) [reply .github/workflows/main.yaml:76]: fixed in: https://github.com/localstack-samples/sample-cdk-kinesis-firehose-redshift/pull/3/commits/50721fca533bcfcb41427357e58866090cc815ec

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**